assets/app/view/game/game_log.rb

```diff
@@ -59,6 +61,124 @@ def render
 
         h(:div, props, children)
       end
+
+      def log
+        has_timestamps = @game.actions.any? { |a| a.is_a?(Engine::Action::Base) && a.created_at }
```

Comment:
why do you check action::base??

Reply:
undone actions are in the actions array as hashes. This is to skip over them.

Reply:
hm maybe you should just fetch the filtered actions

Reply:
Fixed. Split into @actions and @raw_actions array like we talked about.